js/engine.js

```diff
@@ -814,7 +814,20 @@ class Engine {
     const currentValueRaw = self.getStream(stream);
     const currentValue = unitConverter.convert(currentValueRaw, "kg");
 
-    const convertedMax = unitConverter.convert(amount, "kg");
+    let convertedMax;
+
+    // Check if amount is specified in equipment units and add recharge if needed
+    if (amount.getUnits().startsWith("unit")) {
+      // For equipment units, convert to kg and add recharge volume on top
+      const amountInKg = unitConverter.convert(amount, "kg");
+      const rechargeVolume = self._calculateRechargeVolume(null);
```

Comment:
Please check that null is correct here. If so, please change `_calculateRechargeVolume` to work if no parameters are specified such that `_calculateRechargeVolume()` has the same effect as `_calculateRechargeVolume(null)`.